test/integration/test_container_resolvers.py

```diff
@@ -142,17 +157,20 @@ def handle_galaxy_config_kwds(cls, config) -> None:
             config["conda_auto_install"] = True
             config["conda_prefix"] = os.path.join(cls.conda_tmp_prefix, "conda")
 
-    def _clear_container_cache(self):
-        """
-        clear all possibe container caches (ie docker and singularity)
-        """
-        cmd = ["docker", "system", "prune", "--all", "--force", "--volumes"]
-        shell(cmd)
-        if not os.path.exists(self._app.config.container_image_cache_path):
-            return
-        for dirpath, _, files in safe_walk(self._app.config.container_image_cache_path):
-            for f in files:
-                os.unlink(os.path.join(dirpath, f))
+    def _remove_tested_docker_image_from_cache(self):
+        cmd1 = ["docker", "image", "ls", "--quiet", "--filter", f'reference={self.assumptions["run"]["cache_name"]}']
+        image_ids = execute(cmd1)
+        if image_ids:
+            image_id_list = image_ids.splitlines()
+            assert len(image_id_list) == 1
+            cmd2 = ["docker", "image", "rm", "--force", image_id_list[0]]
+            shell(cmd2)
+
+    def _clear_singularity_image_cache(self):
+        if os.path.exists(self._app.config.container_image_cache_path):
+            for dirpath, _, files in safe_walk(self._app.config.container_image_cache_path):
+                for f in files:
+                    os.unlink(os.path.join(dirpath, f))
```

Comment:
Should we also check for n=1 here?

Reply:
I haven't changed this part, this is completely erasing the singularity cache, which lives in a temporary directory, so we are not touching a system cache. The only change is that it's now done at the end of the test, as suggested by Marius.

Reply:
Yes, but analogous to the docker case there should be exactly one image file after each test.

Reply:
This is clean-up phase, not checking that the test was correct. This should be checked as part of the main test.